src/jabs/project/video_manager.py

```diff
@@ -0,0 +1,127 @@
+import json
+import sys
+from pathlib import Path
+
+from jabs.pose_estimation import get_pose_path, get_frames_from_file, open_pose_file
+from jabs.video_reader import VideoReader
+from jabs.video_reader.utilities import get_frame_count
+from .project_paths import ProjectPaths
+from .settings_manager import SettingsManager
+from .video_labels import VideoLabels
+
+
+class VideoManager:
+    def __init__(
+        self,
+        paths: ProjectPaths,
+        settings_manager: SettingsManager,
+        enable_video_check: bool = True,
+    ):
+        """
+        Initialize the VideoManager with paths.
+
+        :param paths: ProjectPaths object containing project directory paths
+        """
+        self._paths = paths
+        self._settings_manager = settings_manager
+        self._videos = []
+        self._total_project_identities = 0
+
+        self._initialize_videos(enable_video_check)
+
+    def _initialize_videos(self, enable_video_check):
+        """Initialize video-related data and perform checks."""
+        self._videos = self.get_videos(self._paths.project_dir)
+        self._videos.sort()
+
+        # self._validate_pose_files()
+        if enable_video_check:
+            self._validate_video_frame_counts()
+
+        self._load_video_metadata()
+
+    @property
+    def videos(self):
+        return self._videos
+
+    @property
+    def total_project_identities(self) -> int:
+        return self._total_project_identities
+
+    def load_video_labels(self, video_name):
+        """
+        load labels for a video from the project directory or from a cached of
+        annotations that have previously been opened and not yet saved
+        :param video_name: filename of the video: string or pathlib.Path
+        :return: initialized VideoLabels object
+        """
+
+        video_filename = Path(video_name).name
+        self.check_video_name(video_filename)
+
+        path = self._paths.annotations_dir / Path(video_filename).with_suffix(".json")
+
+        # if annotations already exist for this video file in the project open
+        # it, otherwise create a new empty VideoLabels
+        if path.exists():
+            with path.open() as f:
+                return VideoLabels.load(json.load(f))
+        else:
+            video_path = self._paths.project_dir / video_filename
+            nframes = get_frame_count(str(video_path))
```

Comment:
This function potentially raises an `IOError`, does it make sense to do any handling of that here or is okay to just implicitly raise it?

Reply:
it's okay to raise it. it should be caught and the GUI will display an alert message indicating why opening the project failed

in the future maybe we want to catch, do some logging, and then reraise